### PR TITLE
(SIMP-4009) Validate against Puppet 5 and OEL

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Tested against Puppet 5
* Tested against OEL

SIMP-4009 #comment Update auditd

### CODE REVIEW

**Comment by jeefberkey** [comment]
:red_circle: https://gitlab.com/simp/pupmod-simp-auditd/pipelines/18605404

@trevor-vaughan can you update the .gitlab.ci.yaml so that the tests run as they're supposed to? I'm not sure how

**Comment by jeefberkey** [comment]
@trevor-vaughan oh, that's easy

**Comment by ayohrling** [comment]
Acceptance: https://gitlab.com/simp/pupmod-simp-auditd/pipelines/25705698

**Comment by ayohrling** [comment]
@jeefberkey Can I get a second set of eyes for a review of this. There were a bunch of conflict resolutions when I merged master over and the changes to the acceptance/module data

**Comment by ayohrling** [comment]
Oracle 6 doesn't seem to have audisp send data to syslog and secure...looking into this now.

**Comment by ayohrling** [comment]
rsyslog doesn't seem to be working at all in OEL6; I can't get anything to log to secure let alone audisp. I even tried concat'ing all of the configurations to the primary rsyslog.conf file just in case it wasn't including the configurations, but it seems to just not be handling the configuration correctly.

Here are the version differences:

[root@oel7 ~]# yum list installed | grep rsyslog
Repository 'chef-current' is missing name in configuration, using id
rsyslog.x86_64                      8.24.0-16.el7_5.4              @ol7_latest
[root@oel7 ~]# yum list installed | grep audit
Repository 'chef-current' is missing name in configuration, using id
audit.x86_64                        2.8.1-3.el7                    @ol7_latest
audit-libs.x86_64                   2.8.1-3.el7                    @ol7_latest


[root@oel6 ~]# yum list installed | grep rsyslog
rsyslog.x86_64                     5.8.10-12.0.1.el6          @public_ol6_latest
[root@oel6 ~]# yum list installed | grep audit
audit.x86_64                       2.4.5-6.el6                @public_ol6_latest
audit-libs.x86_64                  2.4.5-6.el6                @public_ol6_latest

File: manifests/service.pp [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+#
+class auditd::service (
+  $ensure = 'running',
+  $enable = true
+){
   assert_private()
 
   case $facts['os']['name'] {
-    'RedHat','CentOS' : {
+    'RedHat','CentOS','OracleLinux' : {
```
Comment (jeefberkey) [inline manifests/service.pp]: Technically this work is being done by the metadata.json now, but I have just been leaving these as they are for now.

**Comment by ayohrling** [comment]
Waiting for acceptance test to get available runner and execute. Locally the OEL tests now pass with rsyslog being updated for OEL support.

**Comment by jeefberkey** [comment]
:heavy_check_mark: https://gitlab.com/simp/pupmod-simp-auditd/pipelines/25863053

**Comment by jeefberkey** [comment]
it's right this time I swear https://gitlab.com/simp/pupmod-simp-auditd/pipelines/25868354

**Pull request merged**